React/CoreModules/RCTLogBox.mm

```diff
@@ -72,11 +72,95 @@ - (void)show
 
 @end
 
+#else // [TODO(macOS ISS#2323203)
+
+@interface _RCTLogBoxView : NSView
+@end
+
+@implementation _RCTLogBoxView
+- (BOOL)isFlipped
+{
+  return YES;
+}
+- (NSView *)hitTest:(NSPoint)point
+{
+  return self.subviews[0];
+}
+@end
+
+@interface RCTLogBoxWindow : NSWindow
+@end
+
+@implementation RCTLogBoxWindow {
+  RCTSurface *_surface;
+}
+
+- (instancetype)initWithBridge:(RCTBridge *)bridge
+{
+  NSRect minimumFrame = NSMakeRect(0, 0, 600, 800);
+  NSRect maximumFrame = minimumFrame;
+
+  if ((self = [self initWithContentRect:minimumFrame
+                              styleMask:NSWindowStyleMaskTitled
+                                backing:NSBackingStoreBuffered
+                                  defer:YES])) {
+    _surface = [[RCTSurface alloc] initWithBridge:bridge moduleName:@"LogBox" initialProperties:@{}];
+
+    [_surface start];
+    // TODO: It is not entirely clear to me why there needs to be a range. For now we'll use a fixed size.
+    [_surface setMinimumSize:minimumFrame.size maximumSize:maximumFrame.size];
+
+    if (![_surface synchronouslyWaitForStage:RCTSurfaceStageSurfaceDidInitialMounting timeout:1]) {
+      RCTLogInfo(@"Failed to mount LogBox within 1s");
+    }
+
+    self.contentView = (NSView *)_surface.view;
+    self.contentView.autoresizingMask = NSViewWidthSizable | NSViewHeightSizable;
+  }
+  return self;
+}
+
+- (void)hide
+{
+  if (NSApp.modalWindow == self) {
+    [NSApp stopModal];
+  }
+  [self orderOut:nil];
+}
+
+- (void)show
+{
+  if (!RCTRunningInTestEnvironment()) {
+    // Run the modal loop outside of the dispatch queue because it is not reentrant.
+    [self performSelectorOnMainThread:@selector(_showModal) withObject:nil waitUntilDone:NO];
```

Comment:
Wouldn't the iOS version of LogBox have the same reentrant problem?

Reply:
No, this has to do with how `NSWindow` is shown modally vs how it’s done on iOS. See https://github.com/microsoft/react-native-macos/pull/609